collections/_posts/2025-01-18-GradientStepDenoiser.md

```diff
@@ -0,0 +1,218 @@
+
+$$D_\sigma (x) = x - \tau \nabla_x \| x - N_\sigma(x)\|^2$$
+
+Which can be seen as an **explicit gradient step** over the regularization $$g_\sigma (x) = \| x - N_\sigma(x)\|^2$$:
+
+$$D_\sigma (x) = x - \tau \nabla_x g_\sigma (x)$$
+
+This denoiser can then be trained as any other, with an $$L2$$ norm over a range of noise levels (now often called denoising score matching):
+
+$$\mathcal{L}(D_\sigma) = \mathbb{E}_{x \sim p(x), \epsilon_\sigma \sim \mathcal N(0, \sigma^2)} \left[ \| D_\sigma(x + \epsilon_\sigma) - x\right \|^2]$$
+
+**REMARK** : with a bit of algebra and Tweedie's formula we can show that: $$\nabla g_\sigma (x) = - \sigma^2 \nabla \log p_\sigma (x)$$. This offers the nice interpretation of this regularization being tightly related to the **score of the noisy prior distribution** $$ p_\sigma(x)$$.
+
+The authors then show in a series of experiments that this special form of denoiser **does not degrade denoising performances**.
+
+| σ (./255)      | 5      | 15     | 25     | 50     | Time (ms) |
+|---------------|--------|--------|--------|--------|-----------|
+| FFDNet   | 39.95  | 33.53  | 30.84  | 27.54  | 1.9       |
+| DnCNN     | 39.80  | 33.55  | 30.87  | 27.52  | 2.3       |
+| **DRUNet**    | **40.31** | **33.97** | **31.32** | **28.08** | 69.8      |
+|---------------|--------|--------|--------|--------|-----------|
+| DRUNet *light*| 40.19  | 33.89  | 31.25  | 28.00  | 6.3       |
+| **GS-DRUNet**     | 40.26  | 33.90  | 31.26  | 28.01  | 10.4      |
+
+For context, DRUNet is still considered one of the best denoising model.
+The authors achieve roughly similar performances while using a lighter version of the DRUnet (2 residual blocks instead of 4). They even perform slightly better than simply using the exact same architecture, without the gradient-step formulation.
+
+# Plug-and-Play with the Gradient-Step denoiser
+
+The interesting thing about this formulation is that now we have a regularization $$g_\sigma$$ with nice properties. More precisely it is $$L$$-smooth, which is necessary to design convergent first-order algorithms.
+
+I'll skip the details, but the important thing is that now we can devise a **convergent Plug-and-Play** algorithm, with **any neural network** $$N_\sigma$$. The algorithm reads as follows:
+
+$$ x^{n+1} = \text{Prox}_{\tau f} \circ \left( \text{Id} - \tau \lambda \nabla g_\sigma \right) (x^{n})$$
+
+This algorithm is then proven to explicitely decrease the objective function:
+
+$$F(x) = f(x) + g_\sigma (x)$$
+
+which was our initial objective.
+
+**REMARK**: you might notice that it is unusual to take a proximal step on $$f$$ and a gradient step on the regularization $$g$$. This algorithm needs to be formulated as such in order to write the convergence proof, but this is not ideal. In some cases $$f$$ does not admit a closed-form solution for its proximal operator. Subsequent work from the same authors partially fix this issue.
+
+# Experiments
```

Comment:
I was a little lost at the end with the notation, maybe make it appears clearly in the text than GS refers to the gradient step denoiser.